clusterloader2/pkg/prometheus/manifests/dashboards/master-dashboard.dashboard.py

```diff
@@ -0,0 +1,229 @@
+from grafanalib.core import *
+
+DECREASING_ORDER_TOOLTIP = Tooltip(
+    sort=SORT_DESC,
+)
+
+# simple_graph creates a Graph with default dataSource and:
+# * one line (if exprs has type string),
+# * multiple lines (if exprs has type list or tuple)
+
+
+def simple_graph(title, exprs, yAxes=None):
+    if not isinstance(exprs, (list, tuple)):
+        exprs = [exprs]
+    return Graph(
+        title=title,
+        dataSource='$source',
+        span=TOTAL_SPAN // 2,
+        targets=[
+            Target(
+                expr=expr,
+            ) for expr in exprs
+        ],
+        yAxes=yAxes or YAxes(),
+        tooltip=DECREASING_ORDER_TOOLTIP,
+    )
+
+
+dashboard = Dashboard(
+    title="Master dashboard",
+    time=Time("now-30d", "now"),
+    templating=Templating(
+        list=[
+            # Make it possible to use $source as a source.
+            Template(
+                name="source",
+                type="datasource",
+                query="prometheus",
+            )
+        ],
+    ),
+    rows=[
+        Row(
+            title="Clusterloader",
+            height=Pixels(300),
+            panels=[
+                simple_graph(
+                    "Requests",
+                    'sum(rate(apiserver_request_count{client="clusterloader/v0.0.0 (linux/amd64) kubernetes/$Format"}[1m])) by (verb, resource)'),
+                simple_graph(
+                    "API call latency (1s thresholds)",
+                    'apiserver:apiserver_request_latency:histogram_quantile{quantile="0.99", verb!="LIST", verb!="WATCH", verb!="CONNECT"}',
+                    single_y_axis(format=SECONDS_FORMAT)),
+                simple_graph(
+                    "API call latency aggregated (1s thresholds)",
+                    'histogram_quantile(0.99, sum(rate(apiserver_request_duration_seconds_bucket{verb!="LIST", verb!="WATCH", verb!="CONNECT"}[5d])) by (le, resource, verb, scope, subresource))',
```

Comment:
Instead of creating a huge, nested data structure I think we can extract those list to variables and just reference them here. IMO this would makes reading dashboards easier, as there is less nesting.